test_conformance/api/test_semaphores.cpp

```diff
@@ -72,6 +72,75 @@
 
 static const char* source = "__kernel void empty() {}";
 
+/* Infer that signal_event generates a pending signal.
+ * A command that is submitted and not held back by dependencies
+ * moves to READY. If signal_event has no dependencies, polling for submitted
+ * or higher is sufficient.
```

Comment:
This could very well be right, but I'm not quite following this - why is it OK to only check for SUBMITTED if the event has no dependencies?

It might be useful to add a few more words why this is safe, or just to check for RUNNING unconditionally, for simplicity.